### PR TITLE
speed up for gradle

### PR DESCRIPTION
**Pull request by pexcn**
- Enable android build cache
- Increase jvmargs to 2G

### CODE REVIEW

**Comment by Mygod** [comment]
According to Travis, commit d637392 [took 17:46](https://travis-ci.org/shadowsocks/shadowsocks-android/builds/408855809?utm_source=github_status&utm_medium=notification) and this [took 18:27](https://travis-ci.org/shadowsocks/shadowsocks-android/builds/408950311?utm_source=github_status&utm_medium=notification). Maybe the extra cache thing slowed it down? Is it possible to include those cache in travis file as well?

**Comment by pexcn** [comment]
Yes, I add this line:
```
        - $HOME/.android/build-cache/
```

reference: https://docs.travis-ci.com/user/languages/android/#caching

**Comment by pexcn** [comment]
According to travis-ci docs:
https://docs.travis-ci.com/user/languages/android/#gradle-daemon-is-disabled-by-default

but gradle docs was updated, now run daemon by default:
https://docs.gradle.org/current/userguide/gradle_daemon.html#sec:disabling_the_daemon

So we should run gradle as daemon and enable parallel build?
```properties
org.gradle.daemon=true
org.gradle.parallel=true
```

**Comment by Mygod** [comment]
IIRC Travis doc is pretty outdated. You should probably double check if that still applies.

File: gradle.properties [hunk 1]
```diff
@@ -9,11 +9,12 @@
 
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
+android.enableBuildCache=true
```
Comment (aaronhe42) [inline gradle.properties]: this is already default to `true` in AGP 2.3+

**Comment by pexcn** [comment]
I just updated.

**Comment by Mygod** [comment]
Still didn't see any improvements in Travis build after restarting the build. But the code looks fine.

**Comment by pexcn** [comment]
maybe need at least twice build?

**Pull request merged**